pkg/storage/ingest/pusher.go

```diff
@@ -75,14 +79,32 @@ func (c pusherConsumer) pushRequests(ctx context.Context, reqC <-chan parsedReco
 
 		c.processingTimeSeconds.Observe(time.Since(processingStart).Seconds())
 		if err != nil {
-			level.Error(c.l).Log("msg", "failed to push write request; skipping", "err", err)
-			// TODO move distributor's isClientError to a separate package and use that here to swallow only client errors and abort on others
-			continue
+			if !isClientIngesterError(err) {
+				return fmt.Errorf("consuming record at index %d for tenant %s: %w", recordIdx, wr.tenantID, err)
+			}
+			level.Warn(c.l).Log("msg", "consuming write request", "err", err, "user", wr.tenantID)
```

Comment:
[nit] I suggest to write a more user-friendly log message, for example:

```suggestion
			level.Warn(c.l).Log("msg", "a client-side error has been detected while ingesting write request (the request may have been partially ingested)", "err", err, "user", wr.tenantID)
```